File: totvs-marketplace-frontend/src/app/core/services/categoria/categoria.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Categoria, CategoriaDTO } from '../../../models/categoria.model';

@Injectable({
  providedIn: 'root',
})
export class CategoriaService {
  private baseUrl = 'http://localhost:8080/api/categorias';

  constructor(private http: HttpClient) {}

  listarCategorias() {
    return this.http.get<Categoria[]>(this.baseUrl);
  }

  cadastrar(categoria: CategoriaDTO) {
    return this.http.post<Categoria>(this.baseUrl, categoria);
  }

  editar(id: number, categoria: { nome: string }) {
    return this.http.put<Categoria>(`${this.baseUrl}/${id}`, categoria);
  }

  excluir(id: number) {
    return this.http.delete(`${this.baseUrl}/${id}`);
  }
}

File: totvs-marketplace-backend/src/main/java/com/totvs/totvs_marketplace_backend/service/CategoriaService.java
package com.totvs.totvs_marketplace_backend.service;

import java.util.List;
import java.util.Optional;

import com.totvs.totvs_marketplace_backend.model.Categoria;

public interface CategoriaService {
	
	Categoria cadastrar(Categoria categoria);

    Optional<Categoria> buscarPorId(Long id);

    List<Categoria> listarTodas();

    Categoria editar(Long id, Categoria categoria);

    void excluir(Long id);

}
